### PR TITLE
Add proto files for the build event log

### PR DESCRIPTION
**Pull request by tempoz**
- proto files for event log grpc

---

**Version bump**: Minor

### CODE REVIEW

File: proto/eventlog.proto [hunk 1]
```diff
@@ -0,0 +1,65 @@
+syntax = "proto3";
```
Comment (tylerwilliams) [inline proto/eventlog.proto:1]: Can you send just this proto change in a separate PR?

(I believe the chunkstore changes were already approved in https://github.com/buildbuddy-io/buildbuddy/pull/447 -- can you submit that and then remove those changes from this PR too?)

Comment (tempoz) [reply proto/eventlog.proto:1]: Sorry, I forgot to merge the other pr and rebase my branch before pushing

File: proto/eventlog.proto [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  // The request context.
+  context.RequestContext request_context = 1;
+
+  // The invocation identifier itself.
+  string invocation_id = 2;
+	
+	// The range of the event log to retrieve
+	Range range = 3;
+
+}
```
[the remark this answers is not in the document]
Comment (tempoz) [reply proto/eventlog.proto:54]: My plan was only to implement support the chunk ranges at the start but design it this way so that it was easy to add other options later; wdyt? I figured this might be a good idea because if we're letting timeouts form chunks (which we probably want now that the plan is no longer to use the invocation to store the tail) we can end up with really short chunks sometimes, which could be a weird/bad user experience, but still very good for getting started (and also more performant), and then later we can add linewise requests. Maybe this is solvable by including a minimum number of lines in the proto and just retrieve whole chunks until that minimum is satisfied?

[the remark this answers is not in the document]
Comment (tylerwilliams) [reply proto/eventlog.proto]: Ack -- I get what you mean, but I think it's a little premature. HTTP reqs are fast and if we have to do a few back-to-back because we got some small chunks, I don't expect it to be an issue. The server could also mask this by combining consecutive chunks if they are small and just changing the pointer to next-chunk.

Can we start with the simple design and revisit this if it's a problem?

[the remark this answers is not in the document]
Comment (tempoz) [reply proto/eventlog.proto]: yup, no issue with me

File: proto/eventlog.proto [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+option go_package = "proto;eventlog";
+
+message GetEventLogChunkRequest {
+  // The request context.
+  context.RequestContext request_context = 1;
+
+  // The invocation identifier itself.
+  string invocation_id = 2;
+
+  message OptionalChunklID {
```
Comment (tylerwilliams) [inline proto/eventlog.proto:16]: blah blah proto2, proto3, etc. (we talked about this on vc)

can you just make chunk_id a top level-string and remove the optional wrapper?

File: proto/eventlog.proto [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  context.ResponseContext response_context = 1;
+
+  message Chunk {
+    int32 chunk_id = 1;
+    bytes buffer = 2;
+  }
+
+  // The requested chunk
+  Chunk chunk = 2;
+}
```
Comment (tylerwilliams) [inline proto/eventlog.proto:35]: if you add pointers to prev and next you'll have the freedom to change the implementation without changing the clients. Also do things like collapse consecutive chunks without telling the client.

File: proto/eventlog.proto [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  // Optionally specify the chunk to be returned; gets the last chunk if unset
+  OptionalChunkID chunk_id = 3;
+}
+
+message GetEventLogChunkResponse {
+  // The response context.
+  context.ResponseContext response_context = 1;
+
+  message Chunk {
+    int32 chunk_id = 1;
```
Comment (tylerwilliams) [inline proto/eventlog.proto:29]: see prev and next comment below but making this a string gives you some freedom.

(It's basically a pagination token -- and having those be opaque is quite useful! In the past I've written code where a proto containing {int offset, int limit} is encoded and base64d and then used as a pagination token -- then when the server gets that token it can open it up and know where to resume result pagination from.)

**Review by tylerwilliams — CHANGES_REQUESTED** [review]

**Review by tylerwilliams — APPROVED** [review]
LGTM
